edb/schema/delta.py

```diff
@@ -1197,43 +1177,49 @@ def compute_inherited_fields(
 
         return immu.Map(result)
 
-    def _prepare_field_updates(
+    def get_resolved_attribute_value(
         self,
+        attr_name: str,
+        *,
         schema: s_schema.Schema,
         context: CommandContext,
-    ) -> Tuple[s_schema.Schema, Dict[str, Any]]:
-        result = {}
-        metaclass = self.get_schema_metaclass()
+    ) -> Any:
+        raw_value = self.get_attribute_value(attr_name)
+        if raw_value is None:
+            return None
 
-        for op in self.get_subcommands(type=AlterObjectProperty):
-            field = metaclass.get_field(op.property)
+        value = context.get_cached((self, 'attribute', attr_name))
+        if value is None:
+            metaclass = self.get_schema_metaclass()
+            field = metaclass.get_field(attr_name)
             if field is None:
                 raise errors.SchemaDefinitionError(
                     f'got AlterObjectProperty command for '
-                    f'invalid field: {metaclass.__name__}.{op.property}')
+                    f'invalid field: {metaclass.__name__}.{attr_name}')
 
-            val = self._resolve_attr_value(
-                op.new_value, op.property, field, schema)
+            value = self._resolve_attr_value(
+                raw_value, attr_name, field, schema)
 
-            if isinstance(val, s_expr.Expression) and not val.is_compiled():
-                val = self.compile_expr_field(schema, context, field, val)
+            if (isinstance(value, s_expr.Expression)
+                    and not value.is_compiled()):
+                value = self.compile_expr_field(schema, context, field, value)
```

Comment:
Is it safe to cache compiled expressions?

Reply:
Yes, in a sense that both EdgeQL AST and the set of references in the expression are immutable.  IR AST is not safe to cache in general, but it's not being stored currently.